app/backend/src/couchers/models/clusters.py

```diff
@@ -68,32 +73,36 @@ class Cluster(Base):
     discussions_enabled: Mapped[bool] = mapped_column(Boolean, default=True, server_default=expression.true())
     events_enabled: Mapped[bool] = mapped_column(Boolean, default=True, server_default=expression.true())
 
-    slug = column_property(func.slugify(name))
+    slug: Mapped[str] = column_property(func.slugify(name))
 
-    official_cluster_for_node = relationship(
+    official_cluster_for_node: Mapped["Node"] = relationship(
         "Node",
         primaryjoin="and_(Cluster.parent_node_id == Node.id, Cluster.is_official_cluster)",
         back_populates="official_cluster",
         uselist=False,
         viewonly=True,
     )
 
-    parent_node = relationship(
+    parent_node: Mapped["Node"] = relationship(
         "Node", backref="child_clusters", remote_side="Node.id", foreign_keys="Cluster.parent_node_id"
     )
 
-    nodes = relationship("Cluster", backref="clusters", secondary="node_cluster_associations", viewonly=True)
+    nodes: Mapped[list["Cluster"]] = relationship(
+        "Cluster", backref="clusters", secondary="node_cluster_associations", viewonly=True
+    )
     # all pages
-    pages = relationship(
+    pages: DynamicMapped["Page"] = relationship(
         "Page", backref="clusters", secondary="cluster_page_associations", lazy="dynamic", viewonly=True
     )
-    events = relationship("Event", backref="clusters", secondary="cluster_event_associations", viewonly=True)
-    discussions = relationship(
+    events: Mapped["Event"] = relationship(
+        "Event", backref="clusters", secondary="cluster_event_associations", viewonly=True
+    )
+    discussions: Mapped["Discussion"] = relationship(
         "Discussion", backref="clusters", secondary="cluster_discussion_associations", viewonly=True
     )
 
     # includes also admins
-    members = relationship(
+    members: DynamicMapped["User"] = relationship(
```

Comment:
I thought the quoting was only needed if the declaration of User happens further below in the file, but here User is clearly in scope when we get to declarations so I didn't expect needing the `if TYPE_CHECKING` trick, or is it because we have circular references between files?

Reply:
Yes, TYPE_CHECKING is to avoid a circular depdencecy. I'm not 100% sure that there would be one, but it was easier to add TYPE_CHECKING than to check :) 

And since User won't be there at runtime, it has to be a string.